### DISCUSSION TITLE
Terminal printing list of multiple environment variables

### DISCUSSION THREAD

**Discussion by santimchp** [eclipse-cdt-cloud/cdt-cloud#31] [Q&A] [answered]
Hi,
We are using cdt-gdb-vscode extension to debug an application on PolarfireSoC. 
We are able to step through code and debug as expected, however when executing the debug session, a really long list of multiple environment variables are printed on the terminal, package paths, and metadata about Theia IDE. 

Is there a way to somehow configure the debug session to stop these printouts?
Thank you!

The below is our lauch.json file.

```
        "configurations": [{
        "name":"debug-smp",
        "type":"gdbtarget",
        "request":"launch",
        "program": "path/to/executable",
        "gdb": "path/to/gdb",
        "logFile": "path/to/logfile",
        "openGdbConsole": true,
        "initCommands": [
            "set mem inaccessible-by-default off",
            "set $target_riscv=1",
            "set architecture riscv:rv64"
        ],
        "preRunCommands": [
            "thread apply all set $pc=_start",
            "thread apply all set $mie=0",
            "thread apply all set $mip=0",
            "thread apply all set $mstatus=0"
        ],
        "imageAndSymbols": {
            "symbolFileName": "path/to/executable",
            "imageFileName": "path/to/executable"
        },
        "target": {
            "host": "localhost",
            "port": "3333",
            "server": "path/to/openocd",
            "serverParameters": ["--command", "set DEVICE MPFS", "--file", "board/microchip-riscv.cfg", "--command", "set COREID 1"],
            "serverStartupDelay": 5000,
            "cwd": "${workspaceFolder}"
        }
    }]
    }
```

**Comment by jonahgraham** [comment, marked as answer]
A screenshot would help to confirm this, but I think what you are talking about is the environment that is used to launch gdb because you are using `"openGdbConsole": true`. Is it something like:

```
cd /path/to/something ; /usr/bin/env LOTS_OF_ENVS_HERE gdb -ex new-ui\ mi2\ /dev/pts/28 
```

The `openGdbConsole` is obviously a useful feature you don't want to turn off!

It is actually Theia (or VSCode) creating this output because the [adapter](https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/blob/4f6f0fa6e0050a7c50d1d45af4a2c30785f0f14c/src/GDBDebugSession.ts#L455) is using  [runInTerminal](https://microsoft.github.io/debug-adapter-protocol/specification#Reverse_Requests_RunInTerminal) and including the environment variables to use when launching gdb.

The environment it is using is the environment that the debug adapter is running in, which may be different than the environment that the terminal opens in.

Perhaps we could remove the environment as I suspect most of the time it will actually be the same or compatible as what the created terminal will be anyway, at least on Linux (which is all we support so far for `openGdbConsole`). On Windows it is more complicated and I would certainly like to get `openGdbConsole` working on Windows one day.

The ideal place to fix this extra output would be in Theia/VSCode itself when using the `runInTerminal` command. There are other alternatives, such as creating a short shell script that contains the `cd ... ; env ... gdb` part.
